pipeline_dp/pipeline_backend.py

```diff
@@ -470,6 +475,13 @@ def flat_map(self, col, fn, stage_name: str = None):
     def map_tuple(self, col, fn, stage_name: str = None):
         return map(lambda x: fn(*x), col)
 
+    def map_tuple_with_side_inputs(self,
+                                   col,
+                                   fn,
+                                   side_input_cols,
+                                   stage_name: str = None):
+        return map(lambda x: fn(*x, *side_input_cols), col)
```

Comment:
so fn will take all side input arguments

Reply:
Yeah, it looks reasonable for me